src/coap_session.c

```diff
@@ -173,6 +174,12 @@ void coap_session_mfree(coap_session_t *session) {
   if (session->psk_hint)
     coap_free(session->psk_hint);
 
+  CACHE_ITER_SAFE(session->context->cache, cp, ctmp) {
```

Comment:
I suggest using `HASH_ITER` instead to avoid depending on uthash internals.

Reply:
Updated multiple references to CACHE_ITER_SAFE to HASH_ITER.